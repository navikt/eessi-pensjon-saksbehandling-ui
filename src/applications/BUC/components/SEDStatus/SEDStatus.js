import React from 'react'
import PT from 'prop-types'
import classNames from 'classnames'
import { EtikettBase } from 'components/Nav'

import './SEDStatus.css'

const statusList = {
  draft: 'info',
  sent: 'suksess',
  received: 'advarsel',
  unknown: 'fokus'
}

const SEDStatus = (props) => {
  const { className, status, t } = props

  const tagType = Object.prototype.hasOwnProperty.call(statusList, status) ? statusList[status] : statusList['unknown']

  return <EtikettBase className={classNames('a-buc-c-sedstatus', className)} type={tagType}>
    {t('ui:' + props.status)}
  </EtikettBase>
}

SEDStatus.propTypes = {
  className: PT.string,
  status: PT.string.isRequired,
  t: PT.func.isRequired
}

export default SEDStatus
