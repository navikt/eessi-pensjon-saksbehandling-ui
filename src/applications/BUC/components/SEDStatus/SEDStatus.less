.a-buc-c-sedstatus {
    border-radius: 4px;
    text-transform: capitalize;
    font-size: 12px;
    width: 70px;
    align-self: flex-start;
}
