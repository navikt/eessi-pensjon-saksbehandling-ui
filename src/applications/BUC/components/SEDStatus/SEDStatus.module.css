.statusTags {
  &.received {
    background-color: var(--a-lightblue-500) !important;
  }
  &.active {
    background-color: var(--a-purple-200) !important;
    border-color: var(--a-purple-500) !important;
  }
  &.first_sent,
  &.first_received,
  &.first_cancelled {
    background-color: var(--a-gray-400) !important;
  }
}
