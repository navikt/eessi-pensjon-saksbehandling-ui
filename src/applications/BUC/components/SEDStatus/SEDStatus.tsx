import classNames from 'classnames'
import { TPropType } from 'declarations/types.pt'
import Ui from 'eessi-pensjon-ui'
import { Labels } from 'eessi-pensjon-ui/dist/declarations/types'
import PT from 'prop-types'
import React from 'react'
import { T } from 'declarations/types'
import './SEDStatus.css'

export interface SEDStatusProps {
  className ?: string;
  status: string;
  t: T
}

const statusList: Labels = {
  new: 'fokus',
  sent: 'suksess',
  received: 'info',
  cancelled: 'advarsel',
  active: 'advarsel',
  unknown: 'info'
}

const SEDStatus: React.FC<SEDStatusProps> = ({
  className, status, t
}: SEDStatusProps): JSX.Element => {
  const tagType: string = Object.prototype.hasOwnProperty.call(statusList, status) ? statusList[status]! : statusList.unknown!
  return (
    <Ui.Nav.EtikettBase className={classNames('a-buc-c-sedstatus', 'a-buc-c-sedstatus__' + status, className)} type={tagType}>
      {t('buc:status-' + status)}
    </Ui.Nav.EtikettBase>
  )
}

SEDStatus.propTypes = {
  className: PT.string,
  status: PT.string.isRequired,
  t: TPropType.isRequired
}

export default SEDStatus
