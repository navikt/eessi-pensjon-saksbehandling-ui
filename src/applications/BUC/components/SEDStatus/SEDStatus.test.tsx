import { render } from '@testing-library/react'
import SEDStatus, { MyTag, SEDStatusProps } from './SEDStatus'

describe('applications/BUC/components/SEDStatus/SEDStatus', () => {
  let wrapper: any

  const initialMockProps: SEDStatusProps = {
    status: 'new'
  }

  it('Render: match snapshot', () => {
    const { container } = render(<SEDStatus {...initialMockProps} status='new' />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('Render: has proper HTML structure for sent status', () => {
    render(<SEDStatus {...initialMockProps} status='sent' />)
    expect(wrapper.exists(MyTag)).toBeTruthy()
    expect(wrapper.find(MyTag).props().type).toEqual('suksess')
  })
})
