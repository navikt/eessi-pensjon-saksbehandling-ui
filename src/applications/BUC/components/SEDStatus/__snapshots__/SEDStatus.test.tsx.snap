// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications/BUC/components/SEDStatus/SEDStatus Render: match snapshot 1`] = `
.c1.received {
  background-color: var(--a-lightblue-500) !important;
}

.c1.active {
  background-color: var(--a-purple-200) !important;
  border-color: var(--a-purple-500) !important;
}

.c1.first_sent,
.c1.first_received,
.c1.first_cancelled {
  background-color: var(--a-gray-400) !important;
}

ui-fss.g1[id="c0"] {
  content: "c1,";
}

<span
  class="navds-tag c0 c1 new navds-tag--info navds-tag--medium navds-body-short"
>
  buc:status-new
</span>
`;
